### PR TITLE
DPLY-4: Use DbSessionFactory and DbSession to bridge the Hibernate 3 …

### PR DESCRIPTION
**Pull request by ODORA0**
…vs. 4 differences

**Issue worked on:** https://issues.openmrs.org/browse/DPLY-40

### CODE REVIEW

**Comment by ODORA0** [comment]
@dkayiwa Please review this PR, been trying to make sense of the error

**Comment by dkayiwa** [comment]
Did you ask for help on talk?

**Comment by dkayiwa** [comment]
@ODORA0 did you see the travis build failure?

**Comment by ODORA0** [comment]
Yes, been trying to go round it though failing, could I be guided

**Comment by dkayiwa** [comment]
Have you asked on talk?

**Comment by dkayiwa** [comment]
@ODORA0 can your pull request also include a revert of the commit originally made, as per the ticket description?

**Comment by ODORA0** [comment]
@dkayiwa could you please review?

**Comment by dkayiwa** [comment]
@ODORA0 did you see my comment?

**Comment by ODORA0** [comment]
I think I have reverted the previous commit to use the `DbSessionFactory `and `DbSession ` classes

**Comment by dkayiwa** [comment]
Did you take a look at the original commit and confirmed that every green in it has turned into a red in this pull request, and vice versa?

**Comment by ODORA0** [comment]
@dkayiwa I guess it's now been corrected. Please review

File: api/src/main/java/org/openmrs/module/metadatadeploy/builder/ConceptNumericBuilder.java [hunk 1]
```diff
@@ -97,16 +96,7 @@ public ConceptNumericBuilder units(String units) {
     }
 
     public ConceptNumericBuilder precise(Boolean precise) {
-    	try {
-    		((ConceptNumeric) entity).setPrecise(precise);
-    	}
```
[the remark this answers is not in the document]
Comment (ODORA0) [reply api/src/main/java/org/openmrs/module/metadatadeploy/builder/ConceptNumericBuilder.java]: This is part of the revert of the previous commit.

File: api/src/main/java/org/openmrs/module/metadatadeploy/builder/ConceptNumericBuilder.java [hunk 2]
```diff
@@ -97,16 +96,7 @@ public ConceptNumericBuilder units(String units) {
...
-        catch(NoSuchMethodError ex) {
-			try {
-				Method method = ((ConceptNumeric) entity).getClass().getMethod("setAllowDecimal", new Class[] { Boolean.class });
-				method.invoke(((ConceptNumeric) entity), precise);
-			}
-			catch (Exception e) {e.printStackTrace();}
-		}
+        ((ConceptNumeric) entity).setPrecise(precise);
```
Comment (ODORA0) [inline api/src/main/java/org/openmrs/module/metadatadeploy/builder/ConceptNumericBuilder.java]: I set it to it original as seen here

**Comment by dkayiwa** [comment]
Take a second look to confirm that every green in that commit turns out to be a red in this pull request, and vice versa.

**Comment by ODORA0** [comment]
@dkayiwa Just a few unused imports to remove that I am able to note are part of the previous commit,  other than that could you please point out anything else I may not be seeing so I can have them done

**Comment by dkayiwa** [comment]
Take another closer look

**Comment by ODORA0** [comment]
@dkayiwa I think most of the code in that commit have been reverted so as to fit this issue

**Comment by dkayiwa** [comment]
Most, but not all?

**Comment by ODORA0** [comment]
And now, I found some unused imports and white spaces, I think it's good to go?

File: api/src/test/java/org/openmrs/module/metadatadeploy/MetadataUtilsTest.java [hunk 3]
```diff
@@ -189,7 +189,7 @@ public void getForm_shouldThrowExceptionForNonExistent() {
 	@Test
 	public void getLocation_shouldFetchByUuid() {
 		Location unknown = Context.getLocationService().getLocation(1);
-		Assert.assertThat(MetadataUtils.getLocation("dc5c1fcc-0459-4201-bf70-0b90535ba362"), is(unknown));
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/module/metadatadeploy/MetadataUtilsTest.java]: Why did you change this?

Comment (ODORA0) [reply api/src/test/java/org/openmrs/module/metadatadeploy/MetadataUtilsTest.java]: There was an exception thrown that was `org.openmrs.module.metadatadeploy.MissingMetadataException: No such Location with identifier 'dc5c1fcc-0459-4201-bf70-0b90535ba362'` so having that changed to `8d6c993e-c2cc-11de-8d13-0010c6dffd0f` with reference to the location UUID in [this wiki page](https://wiki.openmrs.org/display/docs/UUIDs) did the trick

Comment (dkayiwa) [reply api/src/test/java/org/openmrs/module/metadatadeploy/MetadataUtilsTest.java]: Does that happen only with your changes?

Comment (ODORA0) [reply api/src/test/java/org/openmrs/module/metadatadeploy/MetadataUtilsTest.java]: Yes, that's right
